src/v/cluster/partition.h

```diff
@@ -410,6 +413,8 @@ class partition : public ss::enable_lw_shared_from_this<partition> {
     // exclusive ("write") for enabling/disabling writes
     ssx::rwlock _produce_lock;
 
+    bool _started{false};
```

Comment:
nit: wdyt about naming this some `ready_for_notifications` or something? Or maybe adding a comment about intended use? It feels like this should be used more broadly than it is (e.g. a bunch of calls in this class maybe ought to check _started before doing anything)

Reply:
notification I think is an external construct to the partition state and it depends on the partition being "started", so from that perspective started seems ok to me.